src/App.tsx

```diff
@@ -14,11 +15,204 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+enum SortBy {
+  'None',
+  'Length',
+  'Alphabetically',
+}
+
+interface State {
+  goods: string[];
+  isStart: boolean;
+  isReverse: boolean;
+  isAlphabetically: boolean;
+  isLength: boolean;
+  sortedBy: SortBy;
+  minLength: number;
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: goodsFromServer,
+    isStart: true,
+    isReverse: false,
+    isAlphabetically: false,
+    isLength: false,
+    sortedBy: SortBy.None,
+    minLength: 1,
+  };
+
+  selected = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({
+      minLength: Number(event.currentTarget.value),
+    });
+  };
+
+  sortList = (sortBy: SortBy) => {
+    switch (sortBy) {
+      case SortBy.Alphabetically:
+        this.setState({
+          isAlphabetically: true,
+          isLength: false,
+        });
+        break;
+      case SortBy.Length:
+        this.setState({
+          isAlphabetically: false,
+          isLength: true,
+        });
+
+        break;
+      default:
+    }
+
+    this.setState({
+      sortedBy: sortBy,
+      isReverse: false,
+    });
+  };
+
+  reset = () => {
+    this.setState({
+      isReverse: false,
+      isAlphabetically: false,
+      isLength: false,
+      sortedBy: SortBy.None,
+      minLength: 1,
+    });
+  };
+
+  render() {
+    const {
+      goods,
+      isStart,
+      isReverse,
+      isAlphabetically,
+      isLength,
+      sortedBy,
+      minLength,
+    } = this.state;
+
+    const visibleGoods = goods.filter(item => item.length >= minLength);
+
+    visibleGoods.sort((item1, item2) => {
+      switch (sortedBy) {
+        case SortBy.Alphabetically:
+          return item1.localeCompare(item2);
+        case SortBy.Length:
+          return item1.length - item2.length;
+        default:
+          return 0;
+      }
+    });
```

Comment:
Create a separate function `getVisibleGoods` for this logic
```suggestion
    const visibleGoods = goods.filter(item => item.length >= minLength);

    visibleGoods.sort((item1, item2) => {
      switch (sortedBy) {
        case SortBy.Alphabetically:
          return item1.localeCompare(item2);
        case SortBy.Length:
          return item1.length - item2.length;
        default:
          return 0;
      }
    });
```